pylint/lint/parallel.py

```diff
@@ -3,10 +3,15 @@
 
 import collections
 import functools
+from typing import TYPE_CHECKING, Dict, List, Union
 
 from pylint import reporters
 from pylint.lint.utils import _patch_sys_path
 from pylint.message import Message
+from pylint.typing import CheckerStats
+
+if TYPE_CHECKING:
+    from typing import Counter  # typing.Counter added in Python 3.6.1
```

Comment:
This could also be a `sys.version_info` guard with import from `typing_extensions`. In that case the quotes around `Counter` wouldn't be necessary. Leave that up to you, if you what to change it. Support for `3.6` will end soon anyway, so might not be that important.

Reply:
Do you know if `typing` imports get skipped during runtime? If not it might be better to keep it under the `if TYPE_CHECKING`?

Reply:
No, they are not skipped. Basically they are normal imports. However, since `typing` as well as `typing_extensions` are already imported elsewhere, it shouldn't have much of a performance impact. If any at all.

Reply:
Hmm okay. If you don't have a preference for either I think I will leave it at this.
I actually was thinking about making a checker for imports only being used in typing and a `if TYPE_CHECKING` statement. So this fits nicely!

Reply:
> I actually was thinking about making a checker for imports only being used in typing and a `if TYPE_CHECKING` statement. So this fits nicely!

Just a gut feeling: Don't know how useful such a checker would really be.

Reply:
I thought it might be useful for Python 3.9, where people are more likely to not import from `typing`. For example, in some of my own projects I tend to quickly import from a dependency when I am working in a file that only has a helper function. Although performance effects might be negligible, (I think) it fits with the idea of `pylint` suggesting best practice.
But, this will probably be a discussion for when I make the first PR for this checker 😄